src/schema/provenance_schema.yaml

```diff
@@ -304,6 +308,12 @@ ENTITIES:
       - lab_dataset_id
       - metadata:
         - lab_id
+      - direct_ancestors:
+        - lab_tissue_sample_id
+        - submission_id
+        - lab_dataset_id
+        - metadata:
+          - lab_id
```

Comment:
Considering a Dataset may have multiple parents and they can be Samples or Datasets, for clarity can we add comments to indicate/distinguish which fields belong to what entity type? 

BTW, both Sample and Dataset can have `metadata.lab_id`.

Reply:
It seemed so much clearer to me to get the exclusion fields list for each entity type we recursed through.  But since we're flattening it out in YAML like this, I've added comments.